### ISSUE TITLE
In ratio ding sound doesn't happen when hand jumps to line

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/ratio-and-proportion#370]
**Test device**
Dell
**Operating System**
Win 10
**Browser**
Firefox
**Problem description**
For https://github.com/phetsims/QA/issues/617.
If ratio is reached by dropping a hand close to a line that would be in ratio, without hitting ratio, the hand jumps to the line and is then in ratio. This is good. However, when this happens, the in ratio ding sound does not occur.

**Visuals**

https://user-images.githubusercontent.com/41024075/110706419-0baec980-81b5-11eb-82db-9932cd12d21b.MOV

Troubleshooting information:

<details>

!!!!! DO NOT EDIT !!!!!
Name: ‪Ratio and Proportion‬
URL: https://phet-dev.colorado.edu/html/ratio-and-proportion/1.0.0-rc.3/phet/ratio-and-proportion_all_phet.html
Version: 1.0.0-rc.3 2021-03-03 03:29:06 UTC
Features missing: touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:86.0) Gecko/20100101 Firefox/86.0
Language: en-US
Window: 1280x687
Pixel Ratio: 1.5/1
WebGL: WebGL 1.0
GLSL: WebGL GLSL ES 1.0
Vendor: Mozilla (Mozilla)
Vertex: attribs: 16 varying: 30 uniform: 4096
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 32767x32767
OES_texture_float: true
Dependencies JSON: {}

</details>

**Comment by zepumph** [comment]
In order for the in-proportion sound to play, you have to be interacting with the ratio hand. Thus, when you let go, you no longer have that criteria. So even though the line of code that says "play the in-proportion sound now" is being called, on the very next step it is being silenced because we are no longer interacting with the hand. 

Here is a sloppy, uncommittable patch that demonstrates the issue. If we wait 100ms to say that we are no longer interacting with the hand, then we get the sound a bit. 

Some possible solutions:
1. Somehow have a version of the sound that can complete playing even when no longer interactive.
2. Divide the current sound up into two portions, and detect how far we are through the sound to determine if we shut it off when no longer interacting. This is a question about the capability of the sound library.

**Comment by zepumph** [comment]
So I can very easily reproduce the issue. And @jbphet helped me discuss some of these possible solutions. We went for a time-based accumulator inside the in-proportion SoundGenerator, which ensures that we always play the first ding portion of the sound. This works really well, and I think sounds better in general. It involved some refactoring about how we passed enableControlProperties to the generator, but for the most part this was a simple fix. 

I will bring this up at the design meeting tomorrow since it was a behavioral change in addition to the bug fix, but likely it will just be a confirmation about the current behavior before next RC.

**Comment by zepumph** [comment]
We reviewed this during our meeting today. The behavior was good, and the timing seemed good. @BLFiedler said that he wouldn't want to have it any shorter.

**Comment by zepumph** [comment]
I messed up, and ended up committing a `setTimeout` here, with it gone, my solution doesn't work. I'll keep working on it.

**Comment by zepumph** [comment]
Ok that is now fixed in an actually workable way. Basically I had to make sure that the in proportion sound could play directly when the in-proportion state changed, and not upon next step, because by the next `step`, the ratio hand will have already been toggled user-controlled off. 

Ready for cherry-pick.

**Comment by zepumph** [comment]
Cherry picked.

**Comment by KatieWoe** [comment]
Saw one instance where this still happens.
1. With lines off, get hands close to ratio
2. Turn on lines
3. Click hand to snap it to line and into ratio

This is very minor and may not be worth addressing. Rest of fix seems alright.

https://user-images.githubusercontent.com/41024075/112200709-1ade4000-8bd5-11eb-9730-6e74407217b0.MOV

**Comment by zepumph** [comment]
Really cool edge case @KatieWoe! We built into the in-proportion sound this feature called a "hysteresis threshold". This is where basically, in order to trigger the in proportion sound, it isn't enough to just leave proportion and go back in proportion, but you actually have to leave it by a certain amount, large enough that you don't spam-trigger the in proportion sound too much while trying to get the moving-in-proportion sound. Thus, if you repeat your case but move much more before going back to get close, then you will trigger the sound.

TLDR: this is not a bug, but designed. I'm ready to close this. Feel free to close if that makes sense or continue discussion if you would like.
